src/types/rule.cpp

```diff
@@ -276,6 +276,10 @@ const Meta* Rule::getMetaWithName(const std::string& key) const
  */
 TokenIt Rule::getFirstTokenIt() const
 {
+	if (_mod_private.has_value())
```

Comment:
Wouldn't this assume that there is a specific order in which rule modifiers are specified? But YARA grammar allows to specify both `private global` and `global private`.

Reply:
Yes, I assumed that `private` comes before `global`, which is a mistake, thank you very much for noticing! Will fix that.